### PR TITLE
Reworking documentation paths

### PR DESCRIPTION
**Pull request by arfon**
Replaces #2164

### CODE REVIEW

**Comment by arfon** [comment]
/ cc @aroben for a quick review based on his comment [here](https://github.com/github/linguist/pull/2164#issuecomment-75966654)

File: lib/linguist/vendor.yml [hunk 1]
```diff
@@ -224,19 +228,6 @@
 # Html5shiv
 - (^|/)html5shiv\.js$
 
-# Samples folders
-- ^[Ss]amples/
-
-# LICENSE, README, git config files
-- ^COPYING$
-- LICENSE$
-- License$
```
Comment (aroben) [inline lib/linguist/vendor.yml]: I think "License" won't be matched by `documentation.yml`. Can you add a test for it, and fix it if the test fails?

File: lib/linguist/vendor.yml [hunk 2]
```diff
@@ -224,19 +228,6 @@
...
-- gitattributes$
-- gitignore$
-- gitmodules$
-- ^README$
-- ^readme$
```
Comment (aroben) [inline lib/linguist/vendor.yml]: Ditto for "readme"

**Comment by aroben** [comment]
Looks pretty good! Just two small comments above.

**Comment by arfon** [comment]
Thanks for the review @aroben :zap:

**Pull request merged**